content/en/docs/languages/net/instrumentation.md

```diff
@@ -22,39 +35,244 @@ are covered here as well as the `System.Diagnostics` API.
+        }
+
+        return result;
+    }
+}
+
+```
+
+Replace the content of the Program.cs file with the following code:
+
+```csharp
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddControllers();
+
+var app = builder.Build();
+
+app.MapControllers();
+
+app.Run();
+```
+
+In the `Properties` subdirectory, replace the content of `launchSettings.json` with the following:
+
+```json
+{
+  "$schema": "http://json.schemastore.org/launchsettings.json",
+  "profiles": {
+    "http": {
+      "commandName": "Project",
+      "dotnetRunMessages": true,
+      "launchBrowser": true,
+      "applicationUrl": "http://localhost:8080",
+      "environmentVariables": {
+        "ASPNETCORE_ENVIRONMENT": "Development"
+      }
+    }
+  }
+}
+```
+
+To ensure that it is working, run the application with the following command and
+open <http://localhost:8080/rolldice?rolls=12> in your web browser:
+
+```sh
+dotnet run
+```
+
+You should get a list of 12 numbers in your browser window, for example:
 
-### Initializing tracing
+```text
+[5,6,5,3,6,1,2,5,4,4,2,4]
+```
+
+## Manual instrumentation setup
+
+### Dependencies
 
-There are two main ways to initialize [tracing](/docs/concepts/signals/traces/),
-depending on whether you're using a console app or something that's ASP.NET
-Core-based.
+Install the following OpenTelemetry NuGet packages:
 
-#### Console app
+[OpenTelemetry.Exporter.Console](https://www.nuget.org/packages/OpenTelemetry.Exporter.Console)
 
-To start tracing in a console app, you need to create a tracer provider.
+[OpenTelemetry.Extensions.Hosting](https://www.nuget.org/packages/OpenTelemetry.Extensions.Hosting)
 
-First, ensure that you have the right packages:
 
 ```sh
-dotnet add package OpenTelemetry
 dotnet add package OpenTelemetry.Exporter.Console
+dotnet add package OpenTelemetry.Extensions.Hosting
+```
+
+For ASP.NET Core-based applications, also install the AspNetCore instrumentation package
+
+[OpenTelemetry.Instrumentation.AspNetCore](https://www.nuget.org/packages/OpenTelemetry.Instrumentation.AspNetCore)
+
+```sh
+dotnet add package OpenTelemetry.Instrumentation.AspNetCore
 ```
 
-And then use code like this at the beginning of your program, during any
-important startup operations.
+
+### Initialize the SDK
+
+{{% alert title="Note" color="info" %}} If you’re instrumenting a library, you don't need to initialize the SDK. {{% /alert %}}
+
+It is important to configure an instance of the OpenTelemetry SDK as early as possible in your application.
+
+To initialize the OpenTelemetry SDK for an ASP.NET Core app like in the case of the example app, 
+update the content of the `Program.cs` file with the following code: 
 
 ```csharp
-using OpenTelemetry;
+using OpenTelemetry.Logs;
+using OpenTelemetry.Metrics;
 using OpenTelemetry.Resources;
 using OpenTelemetry.Trace;
 
-// ...
+// Ideally, you will want this name to come from a config file, constants file, etc.
```

Comment:
Because environment variables are the preferred way of configuring Otel, and constants would be a better way to achieve this if it's in code.

That said, I'm not sure commenting the code like this in the getting-started example in the Markdown is worthwhile as it will cause the reader to think about too much.

Reply:
resolving this conversation now as well to merge in this PR, we can have a follow up discussion on that.